### ISSUE TITLE
App crashs if I open the friends list

### ISSUE THREAD

**Issue by syssi** [owntracks/android#984]
* **App build number**: 2.3.0 (23024)
* **Android version**: 9
* **Device**: Xiaomi Mi A2

I'm using a MQTT connection. If a retained owntracks/user/device/info (`_type: card`) message is available but no location message (`_type:location`) the app crashs. If I drop all cards (retained messages of the `info` topic) the view doesn't crash anymore.

```
$ adb logcat
[...]
04-25 12:27:57.683 21452 21452 D AndroidRuntime: Shutting down VM
04-25 12:27:57.683 21452 21452 E AndroidRuntime: FATAL EXCEPTION: main
04-25 12:27:57.683 21452 21452 E AndroidRuntime: Process: org.owntracks.android, PID: 21452
04-25 12:27:57.683 21452 21452 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.owntracks.android.model.messages.MessageLocation.getGeocode()' on a null object reference
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at org.owntracks.android.model.FusedContact.getGeocodedLocation(FusedContact.java:106)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at org.owntracks.android.databinding.UiRowContactBindingImpl.executeBindings(UiRowContactBindingImpl.java:147)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.databinding.ViewDataBinding.executeBindingsInternal(ViewDataBinding.java:473)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.databinding.ViewDataBinding.executePendingBindings(ViewDataBinding.java:445)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at org.owntracks.android.ui.contacts.FusedContactViewHolder.bind(ContactsAdapter.kt:39)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at org.owntracks.android.ui.contacts.ContactsAdapter.onBindViewHolder(ContactsAdapter.kt:22)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at org.owntracks.android.ui.contacts.ContactsAdapter.onBindViewHolder(ContactsAdapter.kt:14)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6118)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6114)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3851)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4404)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:148)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1231)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at com.android.internal.policy.DecorView.onLayout(DecorView.java:753)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.View.layout(View.java:20693)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewGroup.layout(ViewGroup.java:6194)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2799)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2316)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1463)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7190)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.Choreographer.doCallbacks(Choreographer.java:761)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.Choreographer.doFrame(Choreographer.java:696)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:873)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:193)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6718)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-25 12:27:57.683 21452 21452 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by growse** [comment]
Thanks, I'll work on this hopefully tonight and try and get a fix it the next release (2.4.0).

**Comment by growse** [comment]
I think this actually already got fixed in c4a27ef1673732361aa0655bd5c86bd07979eef4. Should be in the next version.
